sympy/solvers/solveset.py

```diff
@@ -978,6 +985,180 @@ def _solveset(f, symbol, domain, _check=False):
+       The target equation.
+    symbol: Symbol
+        The variable for which the eqquation is solved.
+    doamin: Set
+        The domain over which the equation is solved.
+    flags: Dictionary
+        Takes care of the recursive calls.
+
+
+    How is transolve better than _tsolve.
+    ====================================
+
+    1) Improved Output
+
+       The output of many types of equations is much better and easy to
+       understand than the ones computed by _tsolve.
+
+       eg: for 3**(2*x) - 2**(x + 3) transolve would return
+       FiniteSet(-3*log(2)/(-2*log(3) + log(2))), whereas _tsolve
+       would return [-log(2**(3/log(2/9)))]. Both are same but the former
+       increases readability and simplicity.
+
+
+    2) Less Complex API
+
+       transolve's API and its flow is simple to understand. Unlike _tsolve which
+       computes by solving with lots of recursive calls.
+       transolve eases the task by reducing it to a two step procedure as
+       dicussed below.
+
+
+    3) Extensible
+
+       transolve is easily extensible, unlike _tsolve which requires in depth knowledge
+       of the method and adding new class of equation is difficult due to its complex
+       structure.
+       To add a new class of equation in transolve one needs to figure out a way to
+       identify the equation and a generalised way to solve that particular
+       class of eqaution.
+
+
+    How transolve works
+    ===================
+
+    The way transolve solves any transcendental equation is very much
+    different from the old solve way of solving as pointed out above.
+
+    transolve solves equations in two step procedure.
+
+    i)  Identification of the type of equation.
+
+        Helpers are used with heuristics implemented to determine
+        if the equation is of a certain type.
+
+    ii) Invoking the respective helper to solve the equation.
+
+        Once identified what family the equation belongs, respective
+        helpers are called and the equation is solved by either returning
+        the solution or by simplifying the original one to the one that can be
+        handled by _solveset.
+
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.solveset import transolve
+
+    >>> x = symbols('x')
+    >>> transolve(5**(x-3) - 3**(2*x + 1), x, S.Reals)
+    FiniteSet(-log(375)/(-log(5) + 2*log(3)))
+    """
+
+    if 'tsolve_saw' not in flags:
+        flags['tsolve_saw'] = []
+    if f in flags['tsolve_saw']:
+        return ConditionSet(symbol, Eq(f, 0), domain)
+    else:
+        flags['tsolve_saw'].append(f)
+
+    inverter = invert_real if domain.is_subset(S.Reals) else invert_complex
+    lhs, rhs_s = inverter(f, 0, symbol, domain)
+
+    result = ConditionSet(symbol, Eq(f, 0), domain)
+
+    if rhs_s is S.EmptySet:
+        result = S.EmptySet
+    else:
+        rhs = rhs_s.args[0]
+        # do we need a loop for rhs_s?
+        # Can't determine a case as of now.
+        if lhs == symbol:
```

Comment:
This case is added so that even if one tries to solve the equation by calling `transolve`, result can be obtained.
cases like: `2**x  -32` will be handled here.

Reply:
yes